apps/test/unit/blockly/cdoSerializationHelpersTest.js

```diff
@@ -0,0 +1,179 @@
+import {expect} from '../../util/reconfiguredChai';
+import {
+  addPositionsToState,
+  getNewLocation,
+  getCursorYAdjustment,
+  isBlockLocationUnset,
+} from '@cdo/apps/blockly/addons/cdoSerializationHelpers';
+
+describe('addPositionsToState', () => {
+  it('should add x/y values from XML to JSON serialization', () => {
+    const xmlBlocks = [
+      {
+        blockly_block: {
+          id: 'block1',
+          type: 'when_run',
+        },
+        x: 20,
+        y: 20,
+      },
+      {
+        blockly_block: {
+          id: 'block2',
+          type: 'flappy_whenCollideGround',
+        },
+        x: 230,
+        y: 20,
+      },
+      {
+        blockly_block: {
+          id: 'block3',
+          type: 'flappy_whenClick',
+        },
+        x: 20,
+        y: 140,
+      },
+    ];
+
+    const blockIdMap = new Map([
+      ['block1', {type: 'when_run', id: 'block1', x: 0, y: 0}],
```

Comment:
Is there any difference between these three cases? Based on the logic in this function, I'd probably include a case where there's no x/y in the xml. Two cases probably covers it